Jobs delete button

jbalzer12 opened the pull request:
![image](https://user-images.githubusercontent.com/61976072/149121016-b766c8ac-ed4c-4d13-953c-3dc924aced75.png)

Nach Betätigung des Buttons:

![image](https://user-images.githubusercontent.com/61976072/149121101-dc8dfc6d-fcdd-4160-a9a5-22612712355a.png)

@nschnierer Hast du eine Idee, wie man den Button in die gleiche Zeile wie den Job  bekommt, ihn allerdings aus der Verlinkung auf die Ausgabeseite rausnimmt? Aktuell ist der Button nicht Teil des v-list-item, wenn er das wäre, würde er in der gleichen Zeile sein, doch wenn dann der Button gedrückt wird, wird zum Output weitergeleitet und nicht gelöscht.

jbalzer12 committed:
Update api.js

jbalzer12 committed:
delete works

nschnierer commented:
Ich dachte wir machen den Button in die Job Übersicht?

Darüber hinaus sollte der Benutzer nochmal gefragt werden, ob der Job wirklich gelöscht werden soll.

jbalzer12 commented:
> Ich dachte wir machen den Button in die Job Übersicht?
> 
> Darüber hinaus sollte der Benutzer nochmal gefragt werden, ob der Job wirklich gelöscht werden soll.

den Alert kann ich einbauen und der Button ist ja in der Jobsübersicht. In die Ausgabeseite würde ich sagen macht es nicht soviel Sinn den Button zu machen. Darüber sprachen wir beim Download der Log-File

jbalzer12 committed:
alert

jbalzer12 committed:
design fix

jbalzer12 committed:
design update

nschnierer commented on src/features/jobs/JobsView.vue:
```diff
@@ -75,6 +96,12 @@ export default {
+    deleteJob(jobId) {
+      API.deleteJobById(jobId);
+      window.location.reload();
+    },
```
Ein `reload()` ist keine gute Idee. Wir arbeiten in einer Single Page Application (SPA) und hier musst du einfach nur die Daten im Hintergrund laden:

```js
  methods: {
    async deleteJob(jobId) {
      await API.deleteJobById(jobId);
      // reload jobs
      this.$store.dispatch("getJobs");
    },
    // ...
  }
```
Das `await` vor dem löschen ist wichtig, weil wir erst dann die neuen Daten von der API laden dürfen.

jbalzer12 replied:
Ich habe den Vorschlag eingebaut, leider behebt das noch immer nicht das unten beschriebene Problem :(

jbalzer12 committed:
alert built

jbalzer12 committed:
Merge branch 'main' into jobs-delete-button

jbalzer12 committed:
design

jbalzer12 committed:
small corrections

jbalzer12 commented:
![image](https://user-images.githubusercontent.com/61976072/149174696-03e4bb2f-45b2-4ee1-af48-d7a3730b4d3d.png)
![image](https://user-images.githubusercontent.com/61976072/149174745-4b9db3b1-2b7c-49a7-9b40-afe4cf7b449e.png)

@nschnierer : Mit einem Neuladen der Seite wird job initial auf den letzten abgearbeiteten Loops in Zeile 26 (v-for="(job, index) in Object.values(jobs)") gesetzt. Damit wird das Löschen jedes Elements - außer des letzten - nicht korrekt durchgeführt. 
Hast du eine Idee, wie die Jobinformationen an die Listenelemente geknüpft werden können? Im Zweifelsfall müssten wir sonst vielleicht doch darauf zurückgreifen, dass ein Job nur in der Ausgabeübersicht gelöscht werden kann...

nschnierer commented:
> @nschnierer : Mit einem Neuladen der Seite wird job initial auf den letzten abgearbeiteten Loops in Zeile 26 (v-for="(job, index) in Object.values(jobs)") gesetzt. Damit wird das Löschen jedes Elements - außer des letzten - nicht korrekt durchgeführt. Hast du eine Idee, wie die Jobinformationen an die Listenelemente geknüpft werden können? Im Zweifelsfall müssten wir sonst vielleicht doch darauf zurückgreifen, dass ein Job nur in der Ausgabeübersicht gelöscht werden kann...

Muss ich mir mal genauer anschauen. Aber an sich sollte es kein Problem sein, dies korrekt zu aktualisieren.

jbalzer12 committed:
suggestion accepted

jbalzer12 committed:
removed console error "loading used but not known"

https://stackoverflow.com/questions/46401811/vuejs-showing-evaluation-v-if-before-data

jbalzer12 committed:
error removed "duplicate keys detected. that may causes update errors..."

https://stackoverflow.com/questions/51086657/vue-warn-duplicate-keys-detected-x-this-may-cause-an-update-error

jbalzer12 committed:
Update index.js

jbalzer12 commented on src/features/jobs/JobsView.vue:
```diff
@@ -41,10 +41,40 @@
             <v-divider
               v-if="index < Object.keys(jobs).length - 1"
-              :key="job.id"
+              :key="'A' + job.id"
```
https://stackoverflow.com/questions/51086657/vue-warn-duplicate-keys-detected-x-this-may-cause-an-update-error

jbalzer12 commented on src/features/jobs/JobsView.vue:
```diff
@@ -67,16 +97,30 @@
+  },
+  async mounted() {
+    await this.$store.dispatch("getJobs");
+    this.loading = false;
```
https://stackoverflow.com/questions/46401811/vuejs-showing-evaluation-v-if-before-data

jbalzer12 committed:
Update JobsView.vue

nschnierer commented:
Werde mich um das Problem dann kümmern.

nschnierer committed:
Merge remote-tracking branch 'origin/main' into jobs-delete-button

nschnierer committed:
enhance/fix job delete

nschnierer commented:
@jbalzer12 Nun sieht es gut aus :)

Video:
https://user-images.githubusercontent.com/3831579/149923613-a0ec9de9-e838-4f9c-93e9-075043104a0d.mp4

Vercel:
https://digital-peaks-web-aoa-frontend-git-jobs-delet-d655ad-nschnierer.vercel.app
